### PR TITLE
Update vms 2017

### PR DESCRIPTION
**Pull request by vext01**
Let's start looking at this. Please do not merge yet.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -302,12 +311,20 @@ build_v8() {
...
-    cd v8 || exit $?
+    cd ${wrkdir}/v8 || exit $?
     git checkout ${V8_V} || exit $?
     patch -Ep1 < ${PATCH_DIR}/v8.diff || exit $?
-    gclient sync || exit $?
+    cd ${wrkdir}/v8/tools/clang || exit $?
+    patch -Ep1 < ${PATCH_DIR}/v8_clang.diff || exit $?
+
+    cd ${wrkdir}/v8 || exit $?
+    gclient sync --help || exit $?
```
Comment (vext01) [inline build.sh]: ^ can kill this

File: build.sh [hunk 2]
```diff
@@ -477,6 +511,9 @@ build_graal() {
     GRAAL_PATH=`dirname ${OUR_CC}`:${PATH}
     MX="env PATH=${GRAAL_PATH} python2.7 ${wrkdir}/mx/mx.py --java-home ${BOOT_JAVA_HOME}"
 
+    echo "*** Bootstrap JDK8 version:"
```
Comment (vext01) [inline build.sh]: And will kill this

File: build.sh [hunk 3]
```diff
@@ -550,34 +587,61 @@ build_jruby_truffle() {
...
+    cd ${wrkdir}
+    if [ ! -f ${wrkdir}/${CMAKE_TARBALL_BASE}.tar.gz ]; then
+        wget https://cmake.org/files/v3.8/${CMAKE_TARBALL_BASE}.tar.gz || exit $?
+    fi
+    tar xvf ${CMAKE_TARBALL_BASE}.tar.gz || exit $?
+    cd ${wrkdir}/${CMAKE_TARBALL_BASE} || exit $?
+    ./configure --prefix=${CMAKE_INST_DIR} || exit $?
+    make || exit $?
+    make install || exit $?
+}
```
Comment (vext01) [inline build.sh]: This whole hunk can be removed.

File: warmup.krun [hunk 4]
```diff
@@ -124,16 +123,16 @@ SKIP = [
...
     #"*:LuaJIT:*",
     #"*:HHVM:*",
-    #"*:JRubyTruffle:*",
+    #"*:TruffleRuby:*",
     #"*:V8:*",
     #"*:C:*",
 
     # This benchmark is slow *and* crashes
-    "fasta:JRubyTruffle:default-ruby",
+    "fasta:TruffleRuby:default-ruby",
```
Comment (vext01) [inline warmup.krun]: These two benchmarks are still too slow to run. One would add a week, the other a month. Going to look for others that might do the same now.

Comment (vext01) [reply warmup.krun]: nbody/hhvm has a performance regression. Before it took 4 secs per iter, now 10. This means this benchmark/vm pair alone will take a week. Will disable it.

File: build.sh [hunk 5]
```diff
@@ -302,12 +311,17 @@ build_v8() {
     # not master. We use --nohooks now, then once we swich tag we apply our
```
Comment (snim2) [inline build.sh]: `s/swich/switch/g`

File: build.sh [hunk 6]
```diff
@@ -550,34 +581,47 @@ build_jruby_truffle() {
...
+    if [ ! -d ${TRUFFLERUBY_BUILDPACK_DIR} ]; then
+        cd ${wrkdir} && tar zxvf ${TRUFFLERUBY_BUILDPACK_TARBALL} || exit $?
+    fi
+
+    cd ${wrkdir}/truffleruby || exit $?
+
+    # To make a buildpack, you would do:
+    #   env JAVA_HOME=${SYS_JDK8_HOME} mvn -X \
+    #       -Dmaven.repo.local=${TRUFFLERUBY_BUILDPACK_DIR} || exit 1
+    # Then tar up the resultant directory, test it,  and host on archive.org.
```
Comment (snim2) [inline build.sh]: Extra spaces between `,` and `and`

**Comment by snim2** [comment]
Other than those very minor comments, this looks good to me.

**Comment by vext01** [comment]
A quick review of the new commits since last time?

**Comment by snim2** [comment]
New commits look good. Ready to squash?

**Comment by vext01** [comment]
Yep, into one commit again, if that's ok.

**Comment by snim2** [comment]
Sure

**Comment by vext01** [comment]
splat

**Review by snim2 — APPROVED** [review]

**Comment by vext01** [comment]
You can merge both PRs now :)

**Pull request merged**